### PR TITLE
Update random nullifier for split notes

### PR DESCRIPTION
**Pull request by ConstanceBeguier**
To be secure against roadblock attacks, we update the process to obtain a random nullifier for split notes.
Now we have the following formula to evaluate nf_old
- for non split_notes, nf_old = Extract_P([PRF^{nfOrchard}_{nk}(rho_old) + psi_nf) mod q_P] NullifierK + cm_old)
- for split notes, nf_old = Extract_P([PRF^{nfOrchard}_{nk}(rho_old) + psi_nf) mod q_P] NullifierK + cm_old + NullifierL)
where psi_nf is equal to
- psi_old for non split notes
- a random pallas Base element for split notes

The following constraints have been updated into the circuit
- nf_old = nf_old_pub for all notes
- derived_pk_d_old = pk_d_old for all notes
- if split_flag=0, then psi_old = psi_new

### CODE REVIEW

**Review by PaulLaux — APPROVED** [review]
Added minor comment.

Also, please remove comment on line 144 for `bundle.rs`

File: Cargo.toml [hunk 1]
```diff
@@ -29,8 +29,8 @@ blake2b_simd = "1"
 ff = "0.13"
 fpe = "0.6"
 group = { version = "0.13", features = ["wnaf-memuse"] }
-halo2_gadgets = { git = "https://github.com/QED-it/halo2", branch = "zsa1" }
-halo2_proofs = { git = "https://github.com/QED-it/halo2", branch = "zsa1", default-features = false, features = ["batch", "floor-planner-v1-legacy-pdqsort"] }
+halo2_gadgets = { git = "https://github.com/QED-it/halo2", branch = "new_point_from_constant" }
+halo2_proofs = { git = "https://github.com/QED-it/halo2", branch = "new_point_from_constant", default-features = false, features = ["batch", "floor-planner-v1-legacy-pdqsort"] }
```
[the remark this answers is not in the document]
Comment (ConstanceBeguier) [reply Cargo.toml:33]: Done

File: Cargo.toml [hunk 2]
```diff
@@ -54,7 +54,7 @@ plotters = { version = "0.3.0", optional = true }
 [dev-dependencies]
 bridgetree = "0.3"
 criterion = "0.3"
-halo2_gadgets = { git = "https://github.com/QED-it/halo2", branch = "zsa1", features = ["test-dependencies"] }
+halo2_gadgets = { git = "https://github.com/QED-it/halo2", branch = "new_point_from_constant", features = ["test-dependencies"] }
```
[the remark this answers is not in the document]
Comment (ConstanceBeguier) [reply Cargo.toml:57]: Done

File: src/builder.rs [hunk 3]
```diff
@@ -167,26 +167,23 @@ impl SpendInfo {
...
     ///
     /// [TransferZSA]: https://qed-it.github.io/zips/zip-0226.html#split-notes
     fn create_split_spend(&self, rng: &mut impl RngCore) -> Self {
-        let note = self.note;
-        let merkle_path = self.merkle_path.clone();
-
-        let sk = SpendingKey::random(rng);
-        let fvk: FullViewingKey = (&sk).into();
+        let mut note = self.note;
+        note.is_split_note(rng);
```
[the remark this answers is not in the document]
Comment (ConstanceBeguier) [reply src/builder.rs:178]: Done

**Pull request merged**